
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://realmarcin.github.io/linkml-coral/cdm_analysis/CDM_PARQUET_ANALYSIS_REPORT/">
      
      
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.19">
    
    
      
        <title>KBase CDM Parquet Database Analysis Report - linkml-coral</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.7e37652d.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../css/ansi-colours.css">
    
      <link rel="stylesheet" href="../../css/jupyter-cells.css">
    
      <link rel="stylesheet" href="../../css/pandas-dataframe.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#kbase-cdm-parquet-database-analysis-report" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="linkml-coral" class="md-header__button md-logo" aria-label="linkml-coral" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            linkml-coral
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              KBase CDM Parquet Database Analysis Report
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/realmarcin/linkml-coral" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="linkml-coral" class="md-nav__button md-logo" aria-label="linkml-coral" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    linkml-coral
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/realmarcin/linkml-coral" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../elements/index.md" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Schema
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../cdm_store_quickstart/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CDM Store Quick Start
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../about/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    About
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#executive-summary" class="md-nav__link">
    <span class="md-ellipsis">
      Executive Summary
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#1-system-tables-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      1. System Tables Analysis
    </span>
  </a>
  
    <nav class="md-nav" aria-label="1. System Tables Analysis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#11-sys_typedef-118-rows" class="md-nav__link">
    <span class="md-ellipsis">
      1.1 sys_typedef (118 rows)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#12-sys_ddt_typedef-101-rows" class="md-nav__link">
    <span class="md-ellipsis">
      1.2 sys_ddt_typedef (101 rows)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#13-sys_oterm-10594-rows" class="md-nav__link">
    <span class="md-ellipsis">
      1.3 sys_oterm (10,594 rows)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#14-sys_process-142958-rows" class="md-nav__link">
    <span class="md-ellipsis">
      1.4 sys_process (142,958 rows)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#15-sys_process_input-90395-rows" class="md-nav__link">
    <span class="md-ellipsis">
      1.5 sys_process_input (90,395 rows)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#16-sys_process_output-38228-rows" class="md-nav__link">
    <span class="md-ellipsis">
      1.6 sys_process_output (38,228 rows)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-static-data-tables-sdt_" class="md-nav__link">
    <span class="md-ellipsis">
      2. Static Data Tables (sdt_*)
    </span>
  </a>
  
    <nav class="md-nav" aria-label="2. Static Data Tables (sdt_*)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21-naming-conventions" class="md-nav__link">
    <span class="md-ellipsis">
      2.1 Naming Conventions
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22-ontology-term-splitting-pattern" class="md-nav__link">
    <span class="md-ellipsis">
      2.2 Ontology Term Splitting Pattern
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#23-table-inventory-and-sizes" class="md-nav__link">
    <span class="md-ellipsis">
      2.3 Table Inventory and Sizes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#24-foreign-key-relationships" class="md-nav__link">
    <span class="md-ellipsis">
      2.4 Foreign Key Relationships
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#25-schema-differences-from-original-coral" class="md-nav__link">
    <span class="md-ellipsis">
      2.5 Schema Differences from Original CORAL
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-dynamic-data-tables-ddt_" class="md-nav__link">
    <span class="md-ellipsis">
      3. Dynamic Data Tables (ddt_*)
    </span>
  </a>
  
    <nav class="md-nav" aria-label="3. Dynamic Data Tables (ddt_*)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31-ddt_ndarray-brick-index-20-rows" class="md-nav__link">
    <span class="md-ellipsis">
      3.1 ddt_ndarray (Brick Index) - 20 rows
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32-brick-tables-20-tables" class="md-nav__link">
    <span class="md-ellipsis">
      3.2 Brick Tables (20 tables)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-key-schema-patterns" class="md-nav__link">
    <span class="md-ellipsis">
      4. Key Schema Patterns
    </span>
  </a>
  
    <nav class="md-nav" aria-label="4. Key Schema Patterns">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#41-identifier-conventions" class="md-nav__link">
    <span class="md-ellipsis">
      4.1 Identifier Conventions
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#42-ontology-integration" class="md-nav__link">
    <span class="md-ellipsis">
      4.2 Ontology Integration
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#43-provenance-model" class="md-nav__link">
    <span class="md-ellipsis">
      4.3 Provenance Model
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#44-delta-lake-format" class="md-nav__link">
    <span class="md-ellipsis">
      4.4 Delta Lake Format
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5-data-volume-summary" class="md-nav__link">
    <span class="md-ellipsis">
      5. Data Volume Summary
    </span>
  </a>
  
    <nav class="md-nav" aria-label="5. Data Volume Summary">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#51-row-counts-by-category" class="md-nav__link">
    <span class="md-ellipsis">
      5.1 Row Counts by Category
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#52-top-10-tables-by-size" class="md-nav__link">
    <span class="md-ellipsis">
      5.2 Top 10 Tables by Size
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6-comparison-to-original-coral-schema" class="md-nav__link">
    <span class="md-ellipsis">
      6. Comparison to Original CORAL Schema
    </span>
  </a>
  
    <nav class="md-nav" aria-label="6. Comparison to Original CORAL Schema">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#61-structural-changes" class="md-nav__link">
    <span class="md-ellipsis">
      6.1 Structural Changes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#62-new-capabilities" class="md-nav__link">
    <span class="md-ellipsis">
      6.2 New Capabilities
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#63-preserved-features" class="md-nav__link">
    <span class="md-ellipsis">
      6.3 Preserved Features
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7-recommendations-for-linkml-schema" class="md-nav__link">
    <span class="md-ellipsis">
      7. Recommendations for LinkML Schema
    </span>
  </a>
  
    <nav class="md-nav" aria-label="7. Recommendations for LinkML Schema">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#71-schema-organization" class="md-nav__link">
    <span class="md-ellipsis">
      7.1 Schema Organization
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#72-key-patterns-to-implement" class="md-nav__link">
    <span class="md-ellipsis">
      7.2 Key Patterns to Implement
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#73-validation-rules" class="md-nav__link">
    <span class="md-ellipsis">
      7.3 Validation Rules
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#74-documentation-needs" class="md-nav__link">
    <span class="md-ellipsis">
      7.4 Documentation Needs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#8-next-steps" class="md-nav__link">
    <span class="md-ellipsis">
      8. Next Steps
    </span>
  </a>
  
    <nav class="md-nav" aria-label="8. Next Steps">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#81-immediate-actions" class="md-nav__link">
    <span class="md-ellipsis">
      8.1 Immediate Actions
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#82-validation-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      8.2 Validation Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#83-query-examples-to-develop" class="md-nav__link">
    <span class="md-ellipsis">
      8.3 Query Examples to Develop
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#appendix-a-complete-table-listing" class="md-nav__link">
    <span class="md-ellipsis">
      Appendix A: Complete Table Listing
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Appendix A: Complete Table Listing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#system-tables" class="md-nav__link">
    <span class="md-ellipsis">
      System Tables
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#static-data-tables" class="md-nav__link">
    <span class="md-ellipsis">
      Static Data Tables
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#dynamic-data-tables" class="md-nav__link">
    <span class="md-ellipsis">
      Dynamic Data Tables
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#appendix-b-sample-queries" class="md-nav__link">
    <span class="md-ellipsis">
      Appendix B: Sample Queries
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Appendix B: Sample Queries">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#query-1-find-all-samples-from-a-location" class="md-nav__link">
    <span class="md-ellipsis">
      Query 1: Find all samples from a location
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#query-2-trace-assembly-provenance" class="md-nav__link">
    <span class="md-ellipsis">
      Query 2: Trace assembly provenance
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#query-3-find-measurements-for-a-sample" class="md-nav__link">
    <span class="md-ellipsis">
      Query 3: Find measurements for a sample
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="kbase-cdm-parquet-database-analysis-report">KBase CDM Parquet Database Analysis Report</h1>
<p><strong>Database Location:</strong> <code>/Users/marcin/Documents/VIMSS/ENIGMA/KBase/ENIGMA_in_CDM/minio/jmc_coral.db</code></p>
<p><strong>Analysis Date:</strong> 2025-12-01</p>
<hr />
<h2 id="executive-summary">Executive Summary</h2>
<p>The KBase CDM (Common Data Model) database contains <strong>44 tables</strong> organized into three categories:
- <strong>6 system tables</strong> (<code>sys_*</code>): Metadata, type definitions, provenance, and ontology catalogs
- <strong>17 static data tables</strong> (<code>sdt_*</code>): Core scientific entities (samples, assemblies, genomes, etc.)
- <strong>21 dynamic data tables</strong> (<code>ddt_*</code>): Measurement arrays stored as "bricks" with flexible schemas</p>
<p><strong>Total Data Volume:</strong>
- <strong>272,934 rows</strong> across static data tables
- <strong>82.6 million rows</strong> across dynamic data tables (bricks)
- <strong>142,958 process records</strong> tracking provenance
- <strong>10,594 ontology terms</strong> in catalog</p>
<hr />
<h2 id="1-system-tables-analysis">1. System Tables Analysis</h2>
<h3 id="11-sys_typedef-118-rows">1.1 sys_typedef (118 rows)</h3>
<p><strong>Purpose:</strong> Maps CORAL static types to CDM table/column names with constraints</p>
<p><strong>Key Columns:</strong>
- <code>type_name</code>: CORAL entity type (e.g., "Gene", "Bin", "Assembly")
- <code>field_name</code>: Original CORAL field name
- <code>cdm_column_name</code>: Mapped CDM column name (snake_case with prefixes)
- <code>scalar_type</code>: Data type (text, int, float, [text] for arrays)
- <code>pk</code>, <code>upk</code>, <code>fk</code>: Primary key, unique key, foreign key flags
- <code>constraint</code>: Validation patterns or ontology constraints
- <code>units_sys_oterm_id</code>, <code>type_sys_oterm_id</code>: Ontology term references</p>
<p><strong>Key Findings:</strong>
- Defines schema for 18 entity types
- Documents field transformations from CORAL to CDM
- Includes validation constraints and ontology mappings
- Foreign key relationships explicitly defined</p>
<h3 id="12-sys_ddt_typedef-101-rows">1.2 sys_ddt_typedef (101 rows)</h3>
<p><strong>Purpose:</strong> Defines dynamic data table schemas (bricks and microtypes)</p>
<p><strong>Key Columns:</strong>
- <code>ddt_ndarray_id</code>: Brick identifier (e.g., "Brick0000010")
- <code>cdm_column_name</code>: Column name in brick table
- <code>cdm_column_data_type</code>: "variable", "dimension_variable", or "dimension_index"
- <code>scalar_type</code>: Data type (float, object_ref, oterm_ref, text, int, bool)
- <code>dimension_number</code>, <code>variable_number</code>: Position in N-dimensional array
- <code>dimension_oterm_id/name</code>, <code>variable_oterm_id/name</code>: Semantic metadata
- <code>unit_sys_oterm_id/name</code>: Measurement units</p>
<p><strong>Key Findings:</strong>
- 20 brick types defined with varying dimensionality
- Supports complex multi-dimensional arrays (e.g., [209, 52, 3, 3])
- Dimension semantics (Environmental Sample, Molecule, State, Statistic)
- Variable semantics (Concentration, Molecular Weight, etc.)</p>
<h3 id="13-sys_oterm-10594-rows">1.3 sys_oterm (10,594 rows)</h3>
<p><strong>Purpose:</strong> Central ontology term catalog</p>
<p><strong>Key Columns:</strong>
- <code>sys_oterm_id</code>: CURIE (e.g., "ME:0000129", "ENVO:00002041")
- <code>sys_oterm_name</code>: Human-readable term name
- <code>sys_oterm_ontology</code>: Source ontology
- <code>parent_sys_oterm_id</code>: Hierarchical relationships
- <code>sys_oterm_definition</code>: Term definitions
- <code>sys_oterm_synonyms</code>, <code>sys_oterm_links</code>, <code>sys_oterm_properties</code>: Additional metadata</p>
<p><strong>Key Findings:</strong>
- Centralized ontology management
- Primary ontology: <code>context_measurement_ontology</code> (custom ENIGMA ontology)
- Supports ENVO, UO (units), PROCESS, CONTINENT, COUNTRY, MIxS, and other standard ontologies
- Hierarchical structure with parent references</p>
<h3 id="14-sys_process-142958-rows">1.4 sys_process (142,958 rows)</h3>
<p><strong>Purpose:</strong> Provenance tracking for all data transformations</p>
<p><strong>Key Columns:</strong>
- <code>sys_process_id</code>: Unique process identifier
- <code>process_sys_oterm_id/name</code>: Process type (e.g., "Assay Growth", "Sequencing")
- <code>person_sys_oterm_id/name</code>: Person who performed the process
- <code>campaign_sys_oterm_id/name</code>: Research campaign
- <code>sdt_protocol_name</code>: Protocol used
- <code>date_start</code>, <code>date_end</code>: Temporal metadata
- <code>input_objects</code>, <code>output_objects</code>: Arrays of entity references (type:id format)</p>
<p><strong>Key Findings:</strong>
- Complete provenance lineage for all data
- Links processes to protocols, people, and campaigns
- Input/output tracking enables full dependency graph
- Supports multiple inputs and outputs per process</p>
<h3 id="15-sys_process_input-90395-rows">1.5 sys_process_input (90,395 rows)</h3>
<p><strong>Purpose:</strong> Normalized process input relationships</p>
<p><strong>Key Columns:</strong>
- <code>sys_process_id</code>: Process reference
- <code>sdt_&lt;entity&gt;_id</code>: Foreign keys to input entities (Assembly, Bin, Community, Genome, Location, Reads, Sample, Strain, TnSeq_Library)</p>
<p><strong>Key Findings:</strong>
- One row per (process, input) pair
- Denormalized for query performance
- Null columns for entities not involved in specific process</p>
<h3 id="16-sys_process_output-38228-rows">1.6 sys_process_output (38,228 rows)</h3>
<p><strong>Purpose:</strong> Normalized process output relationships</p>
<p><strong>Key Columns:</strong>
- <code>sys_process_id</code>: Process reference
- Entity ID columns for outputs (including <code>ddt_ndarray_id</code> for brick outputs)</p>
<p><strong>Key Findings:</strong>
- Similar structure to sys_process_input
- Includes dynamic data (bricks) as outputs
- Enables forward and backward provenance queries</p>
<hr />
<h2 id="2-static-data-tables-sdt_">2. Static Data Tables (sdt_*)</h2>
<h3 id="21-naming-conventions">2.1 Naming Conventions</h3>
<p><strong>Table Names:</strong> <code>sdt_&lt;snake_case_entity_name&gt;</code>
- Original CORAL "OTU" → <code>sdt_asv</code> (preferred name)
- "TnSeq Library" → <code>sdt_tnseq_library</code></p>
<p><strong>Column Names:</strong>
- Primary keys: <code>sdt_&lt;entity&gt;_id</code> (e.g., <code>sdt_sample_id</code>)
- Names: <code>sdt_&lt;entity&gt;_name</code> (e.g., <code>sdt_sample_name</code>)
- Foreign keys: <code>sdt_&lt;referenced_entity&gt;_name</code> (uses name, not ID for FKs)
- Ontology term splitting: <code>&lt;field&gt;_sys_oterm_id</code> + <code>&lt;field&gt;_sys_oterm_name</code></p>
<h3 id="22-ontology-term-splitting-pattern">2.2 Ontology Term Splitting Pattern</h3>
<p><strong>Original CORAL fields with ontology constraints are split into two columns:</strong></p>
<p>Example from <code>sdt_sample</code>:
- <code>material</code> → <code>material_sys_oterm_id</code> (CURIE) + <code>material_sys_oterm_name</code> (label)
- <code>env_package</code> → <code>env_package_sys_oterm_id</code> + <code>env_package_sys_oterm_name</code></p>
<p>Example from <code>sdt_location</code>:
- <code>continent</code> → <code>continent_sys_oterm_id</code> + <code>continent_sys_oterm_name</code>
- <code>country</code> → <code>country_sys_oterm_id</code> + <code>country_sys_oterm_name</code>
- <code>biome</code> → <code>biome_sys_oterm_id</code> + <code>biome_sys_oterm_name</code>
- <code>feature</code> → <code>feature_sys_oterm_id</code> + <code>feature_sys_oterm_name</code></p>
<p>Example from <code>sdt_reads</code>:
- <code>read_type</code> → <code>read_type_sys_oterm_id</code> + <code>read_type_sys_oterm_name</code>
- <code>sequencing_technology</code> → <code>sequencing_technology_sys_oterm_id</code> + <code>sequencing_technology_sys_oterm_name</code></p>
<p>Example from <code>sdt_community</code>:
- <code>community_type</code> → <code>community_type_sys_oterm_id</code> + <code>community_type_sys_oterm_name</code></p>
<p><strong>Benefits:</strong>
- Enables ontology validation via FK to sys_oterm
- Preserves human-readable labels for queries
- Supports ontology evolution without data migration</p>
<h3 id="23-table-inventory-and-sizes">2.3 Table Inventory and Sizes</h3>
<table>
<thead>
<tr>
<th>Table</th>
<th>Rows</th>
<th>Key Columns</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>sdt_asv</strong></td>
<td>213,044</td>
<td>sdt_asv_id, sdt_asv_name</td>
<td>ASV (Amplicon Sequence Variant) sequences</td>
</tr>
<tr>
<td><strong>sdt_reads</strong></td>
<td>19,307</td>
<td>sdt_reads_id, sdt_reads_name, read_count, read_type_<em>, sequencing_technology_</em>, link</td>
<td>Sequencing reads with ontology terms</td>
</tr>
<tr>
<td><strong>sdt_gene</strong></td>
<td>15,015</td>
<td>sdt_gene_id, sdt_gene_name, sdt_genome_name, contig_number, strand, start, stop, function</td>
<td>Gene annotations</td>
</tr>
<tr>
<td><strong>sdt_genome</strong></td>
<td>6,688</td>
<td>sdt_genome_id, sdt_genome_name, sdt_strain_name, n_contigs, n_features, link</td>
<td>Assembled genomes</td>
</tr>
<tr>
<td><strong>sdt_sample</strong></td>
<td>4,330</td>
<td>sdt_sample_id, sdt_sample_name, sdt_location_name, depth, elevation, date, material_<em>, env_package_</em></td>
<td>Environmental samples with spatial/temporal metadata</td>
</tr>
<tr>
<td><strong>sdt_assembly</strong></td>
<td>3,427</td>
<td>sdt_assembly_id, sdt_assembly_name, sdt_strain_name, n_contigs, link</td>
<td>Sequence assemblies</td>
</tr>
<tr>
<td><strong>sdt_taxon</strong></td>
<td>3,276</td>
<td>sdt_taxon_id, sdt_taxon_name, ncbi_taxid</td>
<td>Taxonomic classifications</td>
</tr>
<tr>
<td><strong>sdt_strain</strong></td>
<td>3,110</td>
<td>sdt_strain_id, sdt_strain_name, sdt_genome_name, derived_from_sdt_strain_name, sdt_gene_names_changed</td>
<td>Microbial strains</td>
</tr>
<tr>
<td><strong>sdt_community</strong></td>
<td>2,209</td>
<td>sdt_community_id, sdt_community_name, community_type_*, sdt_sample_name, parent_sdt_community_name, defined_sdt_strain_names</td>
<td>Microbial communities</td>
</tr>
<tr>
<td><strong>sdt_condition</strong></td>
<td>1,046</td>
<td>sdt_condition_id, sdt_condition_name</td>
<td>Growth conditions</td>
</tr>
<tr>
<td><strong>sdt_location</strong></td>
<td>594</td>
<td>sdt_location_id, sdt_location_name, latitude, longitude, continent_<em>, country_</em>, region, biome_<em>, feature_</em></td>
<td>Sampling locations</td>
</tr>
<tr>
<td><strong>sdt_bin</strong></td>
<td>623</td>
<td>sdt_bin_id, sdt_bin_name, sdt_assembly_name, contigs</td>
<td>Genome bins from metagenomes</td>
</tr>
<tr>
<td><strong>sdt_image</strong></td>
<td>218</td>
<td>sdt_image_id, sdt_image_name, mime_type, size, dimensions, link</td>
<td>Microscopy and other images</td>
</tr>
<tr>
<td><strong>sdt_protocol</strong></td>
<td>42</td>
<td>sdt_protocol_id, sdt_protocol_name, description, link</td>
<td>Experimental protocols</td>
</tr>
<tr>
<td><strong>sdt_dubseq_library</strong></td>
<td>3</td>
<td>sdt_dubseq_library_id, sdt_dubseq_library_name, sdt_genome_name, n_fragments</td>
<td>DubSeq libraries</td>
</tr>
<tr>
<td><strong>sdt_tnseq_library</strong></td>
<td>1</td>
<td>sdt_tnseq_library_id, sdt_tnseq_library_name, sdt_genome_name, primers_model, n_mapped_reads, etc.</td>
<td>TnSeq library</td>
</tr>
<tr>
<td><strong>sdt_enigma</strong></td>
<td>1</td>
<td>sdt_enigma_id</td>
<td>Root entity (database singleton)</td>
</tr>
</tbody>
</table>
<h3 id="24-foreign-key-relationships">2.4 Foreign Key Relationships</h3>
<p><strong>Key Patterns:</strong>
- FKs use <code>_name</code> suffix, not <code>_id</code> (e.g., <code>sdt_location_name</code> in Sample references Location.name)
- Self-referential relationships: Community.parent_sdt_community_name, Strain.derived_from_sdt_strain_name
- Many-to-many via arrays: Community.defined_sdt_strain_names (array of strain names)</p>
<p><strong>Relationship Graph:</strong></p>
<pre><code>Location
  └─ Sample
      ├─ Community
      │   └─ (parent_community, defined_strains)
      └─ Reads
          └─ Assembly
              ├─ Bin
              │   └─ Genome
              │       ├─ Gene
              │       ├─ DubSeq_Library
              │       └─ TnSeq_Library
              └─ Genome

Strain
  ├─ Assembly
  ├─ Genome
  └─ (derived_from relationship)

Image (standalone, linked via sys_process)
Protocol (referenced by sys_process)
Condition (referenced by Community)
Taxon (standalone)
ASV (standalone, linked via bricks)
</code></pre>
<h3 id="25-schema-differences-from-original-coral">2.5 Schema Differences from Original CORAL</h3>
<p><strong>New Fields:</strong>
- <code>link</code>: External URLs/references (Assembly, Genome, Reads, Image, Protocol)
- Ontology term splitting (see section 2.2)
- <code>sdt_</code> prefix on all ID and name columns</p>
<p><strong>Renamed Fields:</strong>
- <code>id</code> → <code>sdt_&lt;entity&gt;_id</code>
- <code>name</code> → <code>sdt_&lt;entity&gt;_name</code>
- Foreign key references use <code>sdt_&lt;entity&gt;_name</code> convention</p>
<p><strong>Split Fields:</strong>
- All ontology-constrained fields split into <code>_sys_oterm_id</code> + <code>_sys_oterm_name</code> pairs</p>
<p><strong>Normalized Structures:</strong>
- Process inputs/outputs denormalized into separate tables for query performance</p>
<hr />
<h2 id="3-dynamic-data-tables-ddt_">3. Dynamic Data Tables (ddt_*)</h2>
<h3 id="31-ddt_ndarray-brick-index-20-rows">3.1 ddt_ndarray (Brick Index) - 20 rows</h3>
<p><strong>Purpose:</strong> Index of all measurement bricks with metadata</p>
<p><strong>Key Columns:</strong>
- <code>ddt_ndarray_id</code>: Brick identifier (e.g., "Brick0000010")
- <code>ddt_ndarray_name</code>: Descriptive name (e.g., "adams_metals_100ws.ndarray")
- <code>ddt_ndarray_description</code>: Full description
- <code>ddt_ndarray_type_sys_oterm_id/name</code>: Brick type (e.g., "Chemical Measurement")
- <code>ddt_ndarray_shape</code>: Array dimensions (e.g., [209, 52, 3, 3])
- <code>ddt_ndarray_dimension_types_sys_oterm_id/name</code>: Dimension semantics (arrays)
- <code>ddt_ndarray_dimension_variable_types_sys_oterm_id/name</code>: Variables per dimension
- <code>ddt_ndarray_variable_types_sys_oterm_id/name</code>: Value variables
- <code>withdrawn_date</code>, <code>superceded_by_ddt_ndarray_id</code>: Versioning metadata</p>
<p><strong>Example Brick:</strong></p>
<pre><code>Brick0000010: adams_metals_100ws.ndarray
  Description: Adams Lab Metals Measurements for 100 Well Survey
  Type: Chemical Measurement (DA:0000005)
  Shape: [209, 52, 3, 3]
  Dimensions:
    1. Environmental Sample (DA:0000042) - variables: Environmental Sample ID
    2. Molecule (ME:0000027) - variables: Molecule from list, Molecular Weight, Algorithm Parameter, Detection Limit
    3. State (ME:0000037) - (no variables listed)
    4. Replicate Series (ME:0000???) - variables: Count Unit
  Values: Concentration (ME:0000129) in micromolar (UO:0000064)
</code></pre>
<h3 id="32-brick-tables-20-tables">3.2 Brick Tables (20 tables)</h3>
<p><strong>Table Naming:</strong> <code>ddt_brick&lt;7-digit-id&gt;</code> (e.g., <code>ddt_brick0000010</code>)</p>
<p><strong>Column Structure:</strong>
- <strong>Dimension variables:</strong> Reference columns (e.g., <code>sdt_sample_name</code>, <code>molecule_from_list_sys_oterm_id</code>)
- <strong>Dimension metadata:</strong> Properties (e.g., <code>molecule_molecular_weight_dalton</code>, <code>state</code>)
- <strong>Value variables:</strong> Measurements (e.g., <code>concentration_micromolar</code>)</p>
<p><strong>Example Brick Schemas:</strong></p>
<p><strong>ddt_brick0000010</strong> (52,884 rows):</p>
<pre><code>- sdt_sample_name (dimension 1: Environmental Sample)
- molecule_from_list_sys_oterm_id (dimension 2: Molecule)
- molecule_from_list_sys_oterm_name
- molecule_molecular_weight_dalton
- molecule_algorithm_parameter
- molecule_detection_limit_micromolar
- state (dimension 3: State)
- replicate_series_count_unit (dimension 4)
- concentration_micromolar (value variable)
</code></pre>
<p><strong>ddt_brick0000452</strong> (113,741 rows):</p>
<pre><code>- sdt_asv_name
- sequence_sequence_type_16s_sequence
</code></pre>
<p><strong>ddt_brick0000080</strong> (98,176 rows):</p>
<pre><code>- sdt_sample_name
- molecule_from_list_sys_oterm_id
- molecule_from_list_sys_oterm_name
- molecule_molecular_weight_dalton
- molecule_presence_molecule_from_list_helium_0
- concentration_statistic_average_parts_per_billion
- concentration_statistic_standard_deviation_parts_per_billion
- detection_limit_parts_per_billion
</code></pre>
<p><strong>Key Findings:</strong>
- Flexible schema supports different measurement types
- Ontology terms embedded for molecules and other categorical dimensions
- Statistical aggregates (mean, std dev) stored in same row
- Detection limits and quality metadata included
- Large data volume: 82+ million rows across 20 bricks</p>
<hr />
<h2 id="4-key-schema-patterns">4. Key Schema Patterns</h2>
<h3 id="41-identifier-conventions">4.1 Identifier Conventions</h3>
<ul>
<li><strong>System IDs:</strong> <code>sys_&lt;entity&gt;_id</code> (e.g., <code>sys_process_id</code>, <code>sys_oterm_id</code>)</li>
<li><strong>Static Data IDs:</strong> <code>sdt_&lt;entity&gt;_id</code> (e.g., <code>sdt_sample_id</code>)</li>
<li><strong>Dynamic Data IDs:</strong> <code>ddt_ndarray_id</code> (for bricks)</li>
<li><strong>All IDs:</strong> Zero-padded sequential (e.g., "Assembly0000001", "Process0122921")</li>
</ul>
<h3 id="42-ontology-integration">4.2 Ontology Integration</h3>
<p><strong>Three Types of Ontology References:</strong>
1. <strong>Direct term usage:</strong> <code>field_sys_oterm_id</code> + <code>field_sys_oterm_name</code> pairs
2. <strong>Metadata annotations:</strong> <code>units_sys_oterm_id</code>, <code>type_sys_oterm_id</code> in sys_typedef
3. <strong>Semantic dimensions:</strong> Dimension and variable ontology terms in sys_ddt_typedef</p>
<p><strong>Ontology Namespaces:</strong>
- <code>ME:*</code> - ENIGMA Measurement/Context Ontology (custom)
- <code>ENVO:*</code> - Environment Ontology
- <code>UO:*</code> - Units of Measurement Ontology
- <code>DA:*</code> - Data Array Ontology (custom)
- <code>PROCESS:*</code> - Process Types (custom)
- <code>ENIGMA:*</code> - People and Campaigns (custom)
- <code>CONTINENT:*</code>, <code>COUNTRY:*</code> - Geography (custom)
- <code>MIxS:*</code> - Minimum Information Standards</p>
<h3 id="43-provenance-model">4.3 Provenance Model</h3>
<p><strong>Hierarchical Tracking:</strong></p>
<pre><code>sys_process (workflow step)
  ├─ inputs (sys_process_input)
  │   ├─ Static entities (sdt_*)
  │   └─ Dynamic arrays (ddt_*)
  ├─ outputs (sys_process_output)
  │   ├─ Static entities (sdt_*)
  │   └─ Dynamic arrays (ddt_*)
  └─ metadata
      ├─ Process type (ontology)
      ├─ Person (ontology)
      ├─ Campaign (ontology)
      ├─ Protocol (reference)
      └─ Dates
</code></pre>
<p><strong>Query Capabilities:</strong>
- Forward lineage: "What was derived from this sample?"
- Backward lineage: "What inputs created this assembly?"
- Process-centric: "Show all sequencing processes by person X"
- Campaign tracking: "All data from Metal Metabolism campaign"</p>
<h3 id="44-delta-lake-format">4.4 Delta Lake Format</h3>
<p><strong>All tables stored as Delta Lake:</strong>
- Directory per table with <code>_delta_log/</code> subdirectory
- Parquet files with UUID naming: <code>part-00000-&lt;uuid&gt;-c000.snappy.parquet</code>
- ACID transactions and time travel support
- Schema evolution capability</p>
<hr />
<h2 id="5-data-volume-summary">5. Data Volume Summary</h2>
<h3 id="51-row-counts-by-category">5.1 Row Counts by Category</h3>
<table>
<thead>
<tr>
<th>Category</th>
<th>Table Count</th>
<th>Total Rows</th>
</tr>
</thead>
<tbody>
<tr>
<td>System Tables</td>
<td>6</td>
<td>242,176</td>
</tr>
<tr>
<td>Static Data</td>
<td>17</td>
<td>272,934</td>
</tr>
<tr>
<td>Dynamic Data (Bricks)</td>
<td>20</td>
<td>82,627,111</td>
</tr>
<tr>
<td><strong>Total</strong></td>
<td><strong>43</strong></td>
<td><strong>83,142,221</strong></td>
</tr>
</tbody>
</table>
<h3 id="52-top-10-tables-by-size">5.2 Top 10 Tables by Size</h3>
<ol>
<li><strong>ddt_brick0000452</strong> - 113,741 rows (ASV sequences)</li>
<li><strong>ddt_brick0000080</strong> - 98,176 rows (molecule concentrations)</li>
<li><strong>ddt_brick0000010</strong> - 52,884 rows (metals measurements)</li>
<li><strong>sdt_asv</strong> - 213,044 rows</li>
<li><strong>sys_process</strong> - 142,958 rows</li>
<li><strong>sys_process_input</strong> - 90,395 rows</li>
<li><strong>sys_process_output</strong> - 38,228 rows</li>
<li><strong>sdt_reads</strong> - 19,307 rows</li>
<li><strong>sdt_gene</strong> - 15,015 rows</li>
<li><strong>sys_oterm</strong> - 10,594 rows</li>
</ol>
<hr />
<h2 id="6-comparison-to-original-coral-schema">6. Comparison to Original CORAL Schema</h2>
<h3 id="61-structural-changes">6.1 Structural Changes</h3>
<p><strong>Table-Level Changes:</strong>
- Added <code>sys_*</code> prefix to all system tables
- Added <code>sdt_*</code> prefix to all static data tables
- Added <code>ddt_*</code> prefix to all dynamic data tables
- Renamed "OTU" → "ASV" (Amplicon Sequence Variant)</p>
<p><strong>Column-Level Changes:</strong>
- All ID columns: <code>id</code> → <code>sdt_&lt;entity&gt;_id</code>
- All name columns: <code>name</code> → <code>sdt_&lt;entity&gt;_name</code>
- All FK columns: <code>&lt;entity&gt;</code> → <code>sdt_&lt;entity&gt;_name</code>
- Ontology fields: Split into <code>_sys_oterm_id</code> + <code>_sys_oterm_name</code> pairs</p>
<h3 id="62-new-capabilities">6.2 New Capabilities</h3>
<p><strong>Enhanced Provenance:</strong>
- Denormalized sys_process_input and sys_process_output tables for query performance
- Process linking to protocols, people, and campaigns</p>
<p><strong>Ontology Management:</strong>
- Centralized sys_oterm catalog
- Foreign key constraints ensure ontology term validity
- Embedded ontology names eliminate joins for display</p>
<p><strong>Measurement Versioning:</strong>
- <code>withdrawn_date</code> and <code>superceded_by_ddt_ndarray_id</code> in brick index
- Supports data quality improvements without deletion</p>
<p><strong>External Linking:</strong>
- <code>link</code> fields for URLs to sequence files, images, protocols</p>
<h3 id="63-preserved-features">6.3 Preserved Features</h3>
<ul>
<li>Core entity model (Sample, Location, Reads, Assembly, Genome, Gene, etc.)</li>
<li>Multi-dimensional array structure (bricks)</li>
<li>Process-based provenance</li>
<li>Flexible ontology annotations</li>
</ul>
<hr />
<h2 id="7-recommendations-for-linkml-schema">7. Recommendations for LinkML Schema</h2>
<h3 id="71-schema-organization">7.1 Schema Organization</h3>
<p><strong>Proposed Module Structure:</strong></p>
<pre><code>linkml_coral_kbase/
  ├── core/
  │   ├── ontology.yaml          # sys_oterm, ontology term references
  │   ├── provenance.yaml        # sys_process, sys_process_input/output
  │   └── types.yaml             # sys_typedef, sys_ddt_typedef
  ├── static_entities/
  │   ├── location_sample.yaml   # Location, Sample
  │   ├── genomics.yaml          # Assembly, Genome, Gene, Bin
  │   ├── microbiology.yaml      # Strain, Community, Taxon
  │   ├── sequencing.yaml        # Reads, ASV
  │   ├── libraries.yaml         # TnSeq_Library, DubSeq_Library
  │   └── metadata.yaml          # Image, Protocol, Condition
  └── dynamic_arrays/
      └── bricks.yaml            # ddt_ndarray, brick metadata
</code></pre>
<h3 id="72-key-patterns-to-implement">7.2 Key Patterns to Implement</h3>
<p><strong>1. Ontology Term Mixin:</strong></p>
<pre><code class="language-yaml">mixins:
  OntologyTermPair:
    attributes:
      sys_oterm_id:
        range: OntologyTerm
        required: true
      sys_oterm_name:
        range: string
        required: true
</code></pre>
<p><strong>2. Entity Naming Convention:</strong></p>
<pre><code class="language-yaml">classes:
  Sample:
    attributes:
      sdt_sample_id:
        identifier: true
        pattern: &quot;^Sample\\d{7}$&quot;
      sdt_sample_name:
        required: true
        unique_key: true
</code></pre>
<p><strong>3. Foreign Key Conventions:</strong></p>
<pre><code class="language-yaml">attributes:
  sdt_location_name:
    range: Location
    inlined: false  # Reference by name, not inline
</code></pre>
<p><strong>4. Process Tracking:</strong></p>
<pre><code class="language-yaml">classes:
  Process:
    attributes:
      sys_process_id:
        identifier: true
      input_objects:
        multivalued: true
        range: string  # Format: &quot;EntityType:EntityID&quot;
      output_objects:
        multivalued: true
        range: string
</code></pre>
<h3 id="73-validation-rules">7.3 Validation Rules</h3>
<p><strong>Required Constraints:</strong>
- ID format patterns (e.g., <code>^Assembly\\d{7}$</code>)
- Latitude range: [-90, 90]
- Longitude range: [-180, 180]
- Ontology term format: <code>^[A-Z_]+:\\d+$</code>
- Date format: ISO 8601</p>
<p><strong>Recommended Enhancements:</strong>
- FK validation against sys_oterm
- Process input/output entity type validation
- Brick dimension/shape validation
- Unit compatibility checks</p>
<h3 id="74-documentation-needs">7.4 Documentation Needs</h3>
<p><strong>Critical Documentation:</strong>
- Ontology term splitting pattern explanation
- FK naming convention (uses <code>_name</code>, not <code>_id</code>)
- Process provenance model
- Brick structure and query patterns
- Delta Lake storage format</p>
<p><strong>Migration Guide:</strong>
- Original CORAL → KBase CDM mapping
- Field renaming dictionary
- Ontology term migration paths</p>
<hr />
<h2 id="8-next-steps">8. Next Steps</h2>
<h3 id="81-immediate-actions">8.1 Immediate Actions</h3>
<ol>
<li><strong>Create LinkML schema modules</strong> following proposed structure</li>
<li><strong>Document ontology term catalog</strong> with term counts and coverage</li>
<li><strong>Map sys_typedef records</strong> to LinkML class definitions</li>
<li><strong>Define process provenance classes</strong> with input/output tracking</li>
<li><strong>Create brick schema templates</strong> from sys_ddt_typedef</li>
</ol>
<h3 id="82-validation-tasks">8.2 Validation Tasks</h3>
<ol>
<li><strong>Cross-validate FKs</strong> between tables</li>
<li><strong>Check ontology term usage</strong> against sys_oterm catalog</li>
<li><strong>Verify process lineage completeness</strong></li>
<li><strong>Test brick dimension consistency</strong></li>
<li><strong>Validate ID format patterns</strong></li>
</ol>
<h3 id="83-query-examples-to-develop">8.3 Query Examples to Develop</h3>
<ol>
<li>Sample → Reads → Assembly → Genome lineage</li>
<li>Provenance chain for a specific measurement brick</li>
<li>All data from a specific geographic location</li>
<li>Strain derivation tree</li>
<li>Process attribution (who did what when)</li>
</ol>
<hr />
<h2 id="appendix-a-complete-table-listing">Appendix A: Complete Table Listing</h2>
<h3 id="system-tables">System Tables</h3>
<ul>
<li>sys_typedef (118 rows)</li>
<li>sys_ddt_typedef (101 rows)</li>
<li>sys_oterm (10,594 rows)</li>
<li>sys_process (142,958 rows)</li>
<li>sys_process_input (90,395 rows)</li>
<li>sys_process_output (38,228 rows)</li>
</ul>
<h3 id="static-data-tables">Static Data Tables</h3>
<ul>
<li>sdt_assembly (3,427 rows)</li>
<li>sdt_asv (213,044 rows)</li>
<li>sdt_bin (623 rows)</li>
<li>sdt_community (2,209 rows)</li>
<li>sdt_condition (1,046 rows)</li>
<li>sdt_dubseq_library (3 rows)</li>
<li>sdt_enigma (1 row)</li>
<li>sdt_gene (15,015 rows)</li>
<li>sdt_genome (6,688 rows)</li>
<li>sdt_image (218 rows)</li>
<li>sdt_location (594 rows)</li>
<li>sdt_protocol (42 rows)</li>
<li>sdt_reads (19,307 rows)</li>
<li>sdt_sample (4,330 rows)</li>
<li>sdt_strain (3,110 rows)</li>
<li>sdt_taxon (3,276 rows)</li>
<li>sdt_tnseq_library (1 row)</li>
</ul>
<h3 id="dynamic-data-tables">Dynamic Data Tables</h3>
<ul>
<li>ddt_ndarray (20 rows)</li>
<li>ddt_brick0000010, 0000072, 0000073, 0000080</li>
<li>ddt_brick0000452, 0000454, 0000457, 0000458, 0000459</li>
<li>ddt_brick0000460, 0000461, 0000462</li>
<li>ddt_brick0000476, 0000477, 0000478, 0000479</li>
<li>ddt_brick0000495, 0000501, 0000507, 0000508</li>
</ul>
<hr />
<h2 id="appendix-b-sample-queries">Appendix B: Sample Queries</h2>
<h3 id="query-1-find-all-samples-from-a-location">Query 1: Find all samples from a location</h3>
<pre><code class="language-sql">SELECT s.*
FROM sdt_sample s
WHERE s.sdt_location_name = 'CPT1'
</code></pre>
<h3 id="query-2-trace-assembly-provenance">Query 2: Trace assembly provenance</h3>
<pre><code class="language-sql">SELECT
  p.sys_process_id,
  p.process_sys_oterm_name,
  pi.sdt_reads_id AS input_reads,
  po.sdt_assembly_id AS output_assembly
FROM sys_process p
JOIN sys_process_input pi ON p.sys_process_id = pi.sys_process_id
JOIN sys_process_output po ON p.sys_process_id = po.sys_process_id
WHERE po.sdt_assembly_id = 'Assembly0000001'
</code></pre>
<h3 id="query-3-find-measurements-for-a-sample">Query 3: Find measurements for a sample</h3>
<pre><code class="language-sql">SELECT
  b.*
FROM ddt_brick0000010 b
WHERE b.sdt_sample_name = 'EU02-D01'
</code></pre>
<hr />
<p><strong>End of Report</strong></p>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["content.tabs.link"], "search": "../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.92b07e13.min.js"></script>
      
    
  </body>
</html>