
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://realmarcin.github.io/linkml-coral/REFINED_QUERY_ANALYSIS/">
      
      
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.19">
    
    
      
        <title>Refined Query Analysis: Isolate Genome Reads - linkml-coral</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.7e37652d.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../css/ansi-colours.css">
    
      <link rel="stylesheet" href="../css/jupyter-cells.css">
    
      <link rel="stylesheet" href="../css/pandas-dataframe.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#refined-query-analysis-isolate-genome-reads" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="linkml-coral" class="md-header__button md-logo" aria-label="linkml-coral" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            linkml-coral
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Refined Query Analysis: Isolate Genome Reads
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/realmarcin/linkml-coral" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="linkml-coral" class="md-nav__button md-logo" aria-label="linkml-coral" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    linkml-coral
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/realmarcin/linkml-coral" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../elements/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Schema
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../cdm_store_quickstart/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CDM Store Quick Start
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../about/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    About
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#query-refinement-excluding-16s-and-metagenome-data" class="md-nav__link">
    <span class="md-ellipsis">
      Query Refinement: Excluding 16S and Metagenome Data
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Query Refinement: Excluding 16S and Metagenome Data">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#read-type-classification" class="md-nav__link">
    <span class="md-ellipsis">
      Read Type Classification
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#naming-patterns" class="md-nav__link">
    <span class="md-ellipsis">
      Naming Patterns
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#comparative-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Comparative Analysis
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Comparative Analysis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#results-by-read-type-50000-reads-threshold" class="md-nav__link">
    <span class="md-ellipsis">
      Results by Read Type (â‰¥50,000 reads threshold)
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Results by Read Type (â‰¥50,000 reads threshold)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-all-reads-no-filter" class="md-nav__link">
    <span class="md-ellipsis">
      1. All Reads (No Filter)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-isolate-genome-reads-only-exclude-16s" class="md-nav__link">
    <span class="md-ellipsis">
      2. Isolate Genome Reads Only (--exclude-16s)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-metagenome16s-reads-only-me0000113" class="md-nav__link">
    <span class="md-ellipsis">
      3. Metagenome/16S Reads Only (ME:0000113)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#key-insights" class="md-nav__link">
    <span class="md-ellipsis">
      Key Insights
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Key Insights">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#isolate-genome-assembly-potential" class="md-nav__link">
    <span class="md-ellipsis">
      Isolate Genome Assembly Potential
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#why-this-matters" class="md-nav__link">
    <span class="md-ellipsis">
      Why This Matters
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#recommended-query" class="md-nav__link">
    <span class="md-ellipsis">
      Recommended Query
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#top-unused-isolate-genome-reads" class="md-nav__link">
    <span class="md-ellipsis">
      Top Unused Isolate Genome Reads
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#implementation-notes" class="md-nav__link">
    <span class="md-ellipsis">
      Implementation Notes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Implementation Notes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#new-query-flags" class="md-nav__link">
    <span class="md-ellipsis">
      New Query Flags
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#provenance-tracking" class="md-nav__link">
    <span class="md-ellipsis">
      Provenance Tracking
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#justfile-convenience-commands" class="md-nav__link">
    <span class="md-ellipsis">
      Justfile Convenience Commands
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#usage-examples" class="md-nav__link">
    <span class="md-ellipsis">
      Usage Examples
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#scientific-impact" class="md-nav__link">
    <span class="md-ellipsis">
      Scientific Impact
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#see-also" class="md-nav__link">
    <span class="md-ellipsis">
      See Also
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="refined-query-analysis-isolate-genome-reads">Refined Query Analysis: Isolate Genome Reads</h1>
<p>Analysis of unused ENIGMA reads filtered by sequencing type to focus on isolate genome assembly candidates.</p>
<h2 id="query-refinement-excluding-16s-and-metagenome-data">Query Refinement: Excluding 16S and Metagenome Data</h2>
<h3 id="read-type-classification">Read Type Classification</h3>
<p>ENIGMA data contains three types of reads:</p>
<table>
<thead>
<tr>
<th>Read Type</th>
<th>ME Code</th>
<th>Description</th>
<th>Primary Use</th>
<th>Count (â‰¥50K)</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Single End</strong></td>
<td>ME:0000114</td>
<td>Isolate genomic reads</td>
<td>Genome assembly</td>
<td>1,840</td>
</tr>
<tr>
<td><strong>Paired End</strong></td>
<td>ME:0000113</td>
<td>Metagenome &amp; 16S reads</td>
<td>Community analysis</td>
<td>9,640</td>
</tr>
<tr>
<td><strong>Generic</strong></td>
<td>ME:0000112</td>
<td>Metatranscriptome reads</td>
<td>Expression analysis</td>
<td>2,938</td>
</tr>
</tbody>
</table>
<h3 id="naming-patterns">Naming Patterns</h3>
<p><strong>Isolate Genome Reads (ME:0000114 - Single End)</strong>:
- Environmental samples with location codes (FW, GW, DP, EU)
- Examples: <code>FW511_7_26_13_02.reads</code>, <code>GW056_87_1_8_13_10.reads</code>
- Trimmed/processed isolates: <code>FW305-C-52-trim.reads_unpaired_fwd</code>
- <strong>Purpose</strong>: Individual bacterial isolate genome sequencing</p>
<p><strong>Metagenome/16S Reads (ME:0000113 - Paired End)</strong>:
- Community sequencing data
- Examples: <code>DP16D_clean.reads</code>, <code>corepilot_170602_16S.reads</code>
- High read counts for deep community coverage
- <strong>Purpose</strong>: Microbial community structure and diversity</p>
<p><strong>Metatranscriptome Reads (ME:0000112 - Generic)</strong>:
- Transcriptome sequencing
- Examples: <code>MPR-WIN1.reads</code>, <code>MT42.reads</code>, <code>MT49.reads</code>
- <strong>Purpose</strong>: Gene expression analysis</p>
<h2 id="comparative-analysis">Comparative Analysis</h2>
<h3 id="results-by-read-type-50000-reads-threshold">Results by Read Type (â‰¥50,000 reads threshold)</h3>
<h4 id="1-all-reads-no-filter">1. All Reads (No Filter)</h4>
<pre><code class="language-bash">just query-unused-reads 50000
</code></pre>
<p><strong>Results:</strong>
- Total 'good' reads: <strong>14,418</strong>
- Used in assemblies: <strong>2,994</strong>
- <strong>UNUSED: 11,608 (79.2%)</strong>
- Total wasted reads: <strong>43.3 BILLION</strong></p>
<h4 id="2-isolate-genome-reads-only-exclude-16s">2. Isolate Genome Reads Only (--exclude-16s)</h4>
<pre><code class="language-bash">uv run python enigma_query.py unused-reads --min-count 50000 --exclude-16s
</code></pre>
<p><strong>Results:</strong>
- Total 'good' reads: <strong>1,840</strong>
- Used in assemblies: <strong>2,994</strong> (includes some lower-count isolate reads)
- <strong>UNUSED: 1,836 (99.8%)</strong>
- Total wasted reads: <strong>230.5 MILLION</strong>
- Avg read count: <strong>125,553</strong>
- Max read count: <strong>4,078,972</strong></p>
<p><strong>Key Finding</strong>: Almost ALL high-quality isolate genome reads remain unused!</p>
<h4 id="3-metagenome16s-reads-only-me0000113">3. Metagenome/16S Reads Only (ME:0000113)</h4>
<pre><code class="language-bash">uv run python enigma_query.py unused-reads --min-count 50000 --read-type ME:0000113
</code></pre>
<p><strong>Results:</strong>
- Total 'good' reads: <strong>9,640</strong>
- Used in assemblies: <strong>2,994</strong>
- <strong>UNUSED: 7,768 (80.6%)</strong>
- Total wasted reads: <strong>25.0 BILLION</strong>
- Avg read count: <strong>3,219,918</strong>
- Max read count: <strong>32,048,946</strong></p>
<h2 id="key-insights">Key Insights</h2>
<h3 id="isolate-genome-assembly-potential">Isolate Genome Assembly Potential</h3>
<p><strong>The Problem:</strong>
- 1,836 high-quality isolate genome read sets are unused
- These represent individual bacterial strains from environmental samples
- Perfect candidates for genome assembly but never assembled</p>
<p><strong>Read Quality:</strong>
- Minimum: 50,003 reads (above threshold)
- Average: 125,553 reads (sufficient for good assembly)
- Maximum: 4,078,972 reads (excellent coverage)</p>
<p><strong>Sample Origins:</strong>
- FW-series: Rifle, CO field site samples
- GW-series: Groundwater samples
- MT-series: Metatranscriptome samples
- Various strain isolates (FW305-C-XX)</p>
<h3 id="why-this-matters">Why This Matters</h3>
<p><strong>For Genome Assembly:</strong>
- Single End reads are specifically for isolate genomes
- 50K+ reads typically provides 20-100X coverage
- Suitable for de novo assembly with modern assemblers
- Each represents a unique environmental bacterial strain</p>
<p><strong>For Resource Utilization:</strong>
- 230 million sequencing reads wasted
- Each sequencing run costs money and time
- These isolates were cultured and sequenced for genome analysis
- Scientific value lost if not assembled</p>
<h2 id="recommended-query">Recommended Query</h2>
<p><strong>For isolate genome assembly candidates:</strong></p>
<pre><code class="language-bash"># Find unused isolate genome reads (â‰¥50K reads)
uv run python enigma_query.py unused-reads --min-count 50000 --exclude-16s

# Higher threshold for best candidates
uv run python enigma_query.py unused-reads --min-count 100000 --exclude-16s

# Export for downstream processing
uv run python enigma_query.py unused-reads --min-count 50000 --exclude-16s --export isolate_genomes_unused.json
</code></pre>
<p><strong>For metagenome/community analysis:</strong></p>
<pre><code class="language-bash"># Find unused metagenome reads
uv run python enigma_query.py unused-reads --min-count 50000 --read-type ME:0000113
</code></pre>
<h2 id="top-unused-isolate-genome-reads">Top Unused Isolate Genome Reads</h2>
<table>
<thead>
<tr>
<th>Rank</th>
<th>Read Set</th>
<th>Count</th>
<th>Category</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>FW305-C-52-trim.reads_unpaired_fwd</td>
<td>4,078,972</td>
<td>very_high</td>
</tr>
<tr>
<td>2</td>
<td>FW305-C-35-trim.reads_unpaired_fwd</td>
<td>3,618,897</td>
<td>very_high</td>
</tr>
<tr>
<td>3</td>
<td>FW305-C-101-trim.reads_unpaired_fwd</td>
<td>2,075,026</td>
<td>very_high</td>
</tr>
<tr>
<td>4</td>
<td>FW305-C-134A-trim.reads_unpaired_fwd</td>
<td>1,976,395</td>
<td>very_high</td>
</tr>
<tr>
<td>5</td>
<td>MT66-cutadapt-trim.reads_unpaired_fwd</td>
<td>1,932,179</td>
<td>very_high</td>
</tr>
</tbody>
</table>
<h2 id="implementation-notes">Implementation Notes</h2>
<h3 id="new-query-flags">New Query Flags</h3>
<p><strong><code>--exclude-16s</code></strong>:
- Filters to only Single End reads (ME:0000114)
- Excludes Paired End (ME:0000113) and Generic (ME:0000112)
- Focuses on isolate genome sequencing data</p>
<p><strong><code>--read-type &lt;TYPE&gt;</code></strong>:
- Explicitly filter by ME code
- Options: ME:0000114 (Single End), ME:0000113 (Paired End), ME:0000112 (Generic)
- More granular control than --exclude-16s</p>
<h3 id="provenance-tracking">Provenance Tracking</h3>
<p>All refined queries are fully tracked:</p>
<pre><code class="language-json">{
  &quot;execution_id&quot;: &quot;12598938c1b0f8c8&quot;,
  &quot;description&quot;: &quot;Find unused reads with &gt;= 50000 raw reads (isolate genome reads only)&quot;,
  &quot;parameters&quot;: {
    &quot;min_count&quot;: 50000,
    &quot;exclude_16s&quot;: true
  },
  &quot;results&quot;: {
    &quot;unused_good_reads&quot;: 1836,
    &quot;total_wasted_reads&quot;: 230514741
  }
}
</code></pre>
<h2 id="justfile-convenience-commands">Justfile Convenience Commands</h2>
<p>Add to <code>project.justfile</code>:</p>
<pre><code class="language-justfile"># Query unused isolate genome reads
[group('data management')]
query-unused-isolates min_count='50000' db='enigma_data.db':
  @echo &quot;ðŸ§¬ Finding unused isolate genome reads (min_count &gt;= {{min_count}})...&quot;
  uv run python enigma_query.py --db {{db}} unused-reads --min-count {{min_count}} --exclude-16s

# Query unused metagenome reads
[group('data management')]
query-unused-metagenomes min_count='50000' db='enigma_data.db':
  @echo &quot;ðŸ¦  Finding unused metagenome/16S reads (min_count &gt;= {{min_count}})...&quot;
  uv run python enigma_query.py --db {{db}} unused-reads --min-count {{min_count}} --read-type ME:0000113
</code></pre>
<h2 id="usage-examples">Usage Examples</h2>
<pre><code class="language-bash"># Basic: Find all unused isolate genome reads â‰¥50K
uv run python enigma_query.py unused-reads --min-count 50000 --exclude-16s

# High quality only: â‰¥100K reads
uv run python enigma_query.py unused-reads --min-count 100000 --exclude-16s

# Export for genome assembly pipeline
uv run python enigma_query.py unused-reads --min-count 50000 --exclude-16s --export candidates.json --top-n 50

# Compare with metagenome data
uv run python enigma_query.py unused-reads --min-count 50000 --read-type ME:0000113

# Using justfile (if commands added)
just query-unused-isolates 50000
just query-unused-metagenomes 50000
</code></pre>
<h2 id="scientific-impact">Scientific Impact</h2>
<p><strong>Potential for New Genome Assemblies:</strong>
- 1,836 isolate genome datasets available
- Average 125K reads = good assembly quality
- Unique environmental bacterial strains
- Could significantly expand ENIGMA genome collection</p>
<p><strong>Resource Recovery:</strong>
- 230 million isolate genome reads recoverable
- Sequencing already paid for
- Just need computational assembly
- High scientific ROI (return on investment)</p>
<h2 id="see-also">See Also</h2>
<ul>
<li><a href="../QUERY_REFERENCE/">QUERY_REFERENCE.md</a> - Complete query command reference</li>
<li><a href="../DEPLOYMENT_PROVENANCE/">DEPLOYMENT_PROVENANCE.md</a> - Provenance tracking guide</li>
<li><a href="../LINKML_STORE_USAGE/">LINKML_STORE_USAGE.md</a> - Database usage guide</li>
</ul>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": ["content.tabs.link"], "search": "../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.92b07e13.min.js"></script>
      
    
  </body>
</html>