
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://realmarcin.github.io/linkml-coral/QUERY_REFERENCE/">
      
      
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.19">
    
    
      
        <title>ENIGMA Query Reference - linkml-coral</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.7e37652d.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../css/ansi-colours.css">
    
      <link rel="stylesheet" href="../css/jupyter-cells.css">
    
      <link rel="stylesheet" href="../css/pandas-dataframe.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#enigma-query-reference" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="linkml-coral" class="md-header__button md-logo" aria-label="linkml-coral" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            linkml-coral
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              ENIGMA Query Reference
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/realmarcin/linkml-coral" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="linkml-coral" class="md-nav__button md-logo" aria-label="linkml-coral" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    linkml-coral
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/realmarcin/linkml-coral" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../elements/index.md" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Schema
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../cdm_store_quickstart/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CDM Store Quick Start
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../about/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    About
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#prerequisites" class="md-nav__link">
    <span class="md-ellipsis">
      Prerequisites
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#main-query-commands" class="md-nav__link">
    <span class="md-ellipsis">
      Main Query Commands
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Main Query Commands">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-find-unused-good-reads" class="md-nav__link">
    <span class="md-ellipsis">
      1. Find Unused "Good" Reads
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-database-statistics" class="md-nav__link">
    <span class="md-ellipsis">
      2. Database Statistics
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-trace-provenance-lineage" class="md-nav__link">
    <span class="md-ellipsis">
      3. Trace Provenance Lineage
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-find-entities-by-criteria" class="md-nav__link">
    <span class="md-ellipsis">
      4. Find Entities by Criteria
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#common-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      Common Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Common Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#compare-read-types" class="md-nav__link">
    <span class="md-ellipsis">
      Compare Read Types
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#compare-different-thresholds" class="md-nav__link">
    <span class="md-ellipsis">
      Compare Different Thresholds
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#batch-export-multiple-queries" class="md-nav__link">
    <span class="md-ellipsis">
      Batch Export Multiple Queries
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#generate-summary-report" class="md-nav__link">
    <span class="md-ellipsis">
      Generate Summary Report
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#python-api-usage" class="md-nav__link">
    <span class="md-ellipsis">
      Python API Usage
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output-file-naming-conventions" class="md-nav__link">
    <span class="md-ellipsis">
      Output File Naming Conventions
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#database-information" class="md-nav__link">
    <span class="md-ellipsis">
      Database Information
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#performance-notes" class="md-nav__link">
    <span class="md-ellipsis">
      Performance Notes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#troubleshooting" class="md-nav__link">
    <span class="md-ellipsis">
      Troubleshooting
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#provenance-tracking" class="md-nav__link">
    <span class="md-ellipsis">
      Provenance Tracking
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Provenance Tracking">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#what-is-tracked" class="md-nav__link">
    <span class="md-ellipsis">
      What is Tracked
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#view-execution-history" class="md-nav__link">
    <span class="md-ellipsis">
      View Execution History
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#provenance-records" class="md-nav__link">
    <span class="md-ellipsis">
      Provenance Records
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#reproducing-results" class="md-nav__link">
    <span class="md-ellipsis">
      Reproducing Results
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#see-also" class="md-nav__link">
    <span class="md-ellipsis">
      See Also
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="enigma-query-reference">ENIGMA Query Reference</h1>
<p>Quick reference for querying the ENIGMA database using linkml-store.</p>
<h2 id="prerequisites">Prerequisites</h2>
<ol>
<li><strong>Load the data first</strong> (one-time setup):
   <code>bash
   just load-store</code>
   This creates <code>enigma_data.db</code> (10-13 MB) with all ENIGMA data.</li>
</ol>
<h2 id="main-query-commands">Main Query Commands</h2>
<h3 id="1-find-unused-good-reads">1. Find Unused "Good" Reads</h3>
<p><strong>Question:</strong> <em>How many reads with significant raw read counts were NOT used in assemblies?</em></p>
<pre><code class="language-bash"># Using justfile (recommended)
just query-unused-reads 50000

# Or directly
uv run python enigma_query.py --db enigma_data.db unused-reads --min-count 50000
</code></pre>
<p><strong>Refined Queries for Isolate Genomes:</strong></p>
<pre><code class="language-bash"># Isolate genome reads only (exclude 16S/metagenome data)
just query-unused-isolates 50000

# Or directly
uv run python enigma_query.py unused-reads --min-count 50000 --exclude-16s

# Specific read type filter (e.g., Single End reads)
uv run python enigma_query.py unused-reads --min-count 50000 --read-type ME:0000114

# Metagenome/16S reads only
just query-unused-metagenomes 50000

# Or directly
uv run python enigma_query.py unused-reads --min-count 50000 --read-type ME:0000113
</code></pre>
<p><strong>Read Type Codes:</strong>
- <code>ME:0000114</code> - Single End Read (isolate genome sequencing)
- <code>ME:0000113</code> - Paired End Read (metagenome/16S sequencing)
- <code>ME:0000112</code> - Generic Read Type (metatranscriptome)</p>
<p><strong>Output:</strong>
- Summary statistics (total, used, unused, utilization rate)
- Statistics about unused reads (min, max, avg, total wasted)
- Top N unused reads by count (default: 20)</p>
<p><strong>Export to JSON:</strong></p>
<pre><code class="language-bash"># Export all results
uv run python enigma_query.py unused-reads --min-count 50000 --export results.json

# Export isolate genome candidates
uv run python enigma_query.py unused-reads --min-count 50000 --exclude-16s --export isolate_genomes.json

# Export with custom top-N
uv run python enigma_query.py unused-reads --min-count 50000 --top-n 50 --export results.json
</code></pre>
<p><strong>JSON Structure:</strong></p>
<pre><code class="language-json">{
  &quot;query&quot;: &quot;unused_reads&quot;,
  &quot;parameters&quot;: {
    &quot;min_count&quot;: 50000
  },
  &quot;summary&quot;: {
    &quot;min_count_threshold&quot;: 50000,
    &quot;total_good_reads&quot;: 14418,
    &quot;reads_used_in_assemblies&quot;: 2994,
    &quot;unused_good_reads&quot;: 11608,
    &quot;utilization_rate&quot;: 0.2077,
    &quot;unused_stats&quot;: {
      &quot;min_count&quot;: 50003,
      &quot;max_count&quot;: 549479714,
      &quot;avg_count&quot;: 3729317.79,
      &quot;total_wasted_reads&quot;: 43289920880
    }
  },
  &quot;results&quot;: [
    {
      &quot;reads_id&quot;: &quot;Reads0000001&quot;,
      &quot;reads_name&quot;: &quot;FW511_7_26_13_02.reads&quot;,
      &quot;reads_read_count&quot;: 76138,
      &quot;reads_read_type&quot;: &quot;ME:0000114&quot;,
      &quot;reads_sequencing_technology&quot;: &quot;ME:0000117&quot;,
      &quot;reads_link&quot;: &quot;https://narrative.kbase.us/#dataview/...&quot;,
      &quot;read_count_category&quot;: &quot;high&quot;
    }
    // ... 11,607 more results
  ]
}
</code></pre>
<h3 id="2-database-statistics">2. Database Statistics</h3>
<p><strong>View comprehensive database statistics:</strong></p>
<pre><code class="language-bash">just query-stats

# Or directly
uv run python enigma_query.py stats
</code></pre>
<p><strong>Output:</strong>
- Record counts per collection
- Read count distributions
- Process statistics
- Read utilization rates</p>
<h3 id="3-trace-provenance-lineage">3. Trace Provenance Lineage</h3>
<p><strong>Trace the complete lineage for an entity:</strong></p>
<pre><code class="language-bash"># Trace an assembly
just query-lineage Assembly Assembly0000001

# Trace a genome
just query-lineage Genome Genome0000001

# Or directly
uv run python enigma_query.py lineage Assembly Assembly0000001
</code></pre>
<p><strong>Output:</strong>
- Process chain length
- Input reads used
- Input samples
- Full provenance tree</p>
<p><strong>Export to JSON:</strong></p>
<pre><code class="language-bash">uv run python enigma_query.py lineage Assembly Assembly0000001 --export assembly_lineage.json
</code></pre>
<h3 id="4-find-entities-by-criteria">4. Find Entities by Criteria</h3>
<p><strong>Search for entities using key=value filters:</strong></p>
<pre><code class="language-bash"># Find high-count reads
just query-find Reads --query read_count_category=high

# Find assemblies by strain
uv run python enigma_query.py find Assembly --query assembly_strain=FW305-37 --limit 10

# Multiple criteria
uv run python enigma_query.py find Reads --query read_count_category=very_high --limit 50
</code></pre>
<p><strong>Export results:</strong></p>
<pre><code class="language-bash">uv run python enigma_query.py find Reads --query read_count_category=very_high --export high_reads.json
</code></pre>
<h2 id="common-workflows">Common Workflows</h2>
<h3 id="compare-read-types">Compare Read Types</h3>
<pre><code class="language-bash"># Compare utilization across different read types
echo &quot;=== All Reads ===&quot;
just query-unused-reads 50000

echo &quot;=== Isolate Genome Reads Only ===&quot;
just query-unused-isolates 50000

echo &quot;=== Metagenome/16S Reads Only ===&quot;
just query-unused-metagenomes 50000

# Or use the comparison script
./read_type_comparison.sh
</code></pre>
<p><strong>Results at 50K threshold:</strong>
- All reads: 14,418 total, 11,608 unused (79.2%)
- Isolate genomes: 1,840 total, 1,836 unused (99.8%!)
- Metagenomes: 9,640 total, 7,768 unused (80.6%)</p>
<h3 id="compare-different-thresholds">Compare Different Thresholds</h3>
<pre><code class="language-bash"># Compare utilization at different read count thresholds
for threshold in 10000 50000 100000 500000; do
  echo &quot;=== Threshold: $threshold ===&quot;
  uv run python enigma_query.py unused-reads --min-count $threshold --export unused_${threshold}.json
done

# For isolate genomes specifically
for threshold in 50000 100000 200000; do
  echo &quot;=== Isolate Genomes, Threshold: $threshold ===&quot;
  uv run python enigma_query.py unused-reads --min-count $threshold --exclude-16s --export isolates_unused_${threshold}.json
done
</code></pre>
<h3 id="batch-export-multiple-queries">Batch Export Multiple Queries</h3>
<pre><code class="language-bash"># Export unused reads at different thresholds
just query-unused-reads 10000 &gt; unused_10k.txt
just query-unused-reads 50000 &gt; unused_50k.txt
just query-unused-reads 100000 &gt; unused_100k.txt

# Export by read type
just query-unused-isolates 50000 &gt; unused_isolates_50k.txt
just query-unused-metagenomes 50000 &gt; unused_metagenomes_50k.txt

# Export with JSON
uv run python enigma_query.py unused-reads --min-count 10000 --export unused_10k.json
uv run python enigma_query.py unused-reads --min-count 50000 --exclude-16s --export isolates_50k.json
uv run python enigma_query.py unused-reads --min-count 50000 --read-type ME:0000113 --export metagenomes_50k.json
</code></pre>
<h3 id="generate-summary-report">Generate Summary Report</h3>
<pre><code class="language-bash"># Create comprehensive analysis report
{
  echo &quot;# ENIGMA Data Analysis Report&quot;
  echo &quot;Generated: $(date)&quot;
  echo &quot;&quot;
  echo &quot;## Database Statistics&quot;
  uv run python enigma_query.py stats
  echo &quot;&quot;
  echo &quot;## Unused Reads Analysis (50K+ threshold)&quot;
  uv run python enigma_query.py unused-reads --min-count 50000
} &gt; enigma_analysis_report.txt
</code></pre>
<h2 id="python-api-usage">Python API Usage</h2>
<p>For programmatic access in scripts or notebooks:</p>
<pre><code class="language-python">from query_enigma_provenance import ENIGMAProvenanceQuery

# Initialize
query = ENIGMAProvenanceQuery(&quot;enigma_data.db&quot;)

# Get unused reads with details (all reads)
unused_reads, summary = query.get_unused_reads(min_count=50000, return_details=True)

print(f&quot;Unused reads: {summary['unused_good_reads']}&quot;)
print(f&quot;Utilization rate: {summary['utilization_rate']:.1%}&quot;)
print(f&quot;Total wasted reads: {summary['unused_stats']['total_wasted_reads']:,}&quot;)

# Get unused isolate genome reads only
isolate_reads, isolate_summary = query.get_unused_reads(
    min_count=50000,
    return_details=True,
    exclude_16s=True  # Filter to Single End reads only
)

print(f&quot;Unused isolate genomes: {isolate_summary['unused_good_reads']}&quot;)
print(f&quot;Average read count: {isolate_summary['unused_stats']['avg_count']:,.0f}&quot;)

# Get unused reads by specific type
metagenome_reads, meta_summary = query.get_unused_reads(
    min_count=50000,
    return_details=True,
    read_type='ME:0000113'  # Paired End reads
)

print(f&quot;Unused metagenomes: {meta_summary['unused_good_reads']}&quot;)

# Get assembly lineage
lineage = query.get_assembly_lineage(&quot;Assembly0000001&quot;)
print(f&quot;Process steps: {lineage['process_count']}&quot;)
print(f&quot;Input reads: {len(lineage['input_reads'])}&quot;)

# Get all reads by category
high_count_reads = query.get_all_reads(category='very_high')
print(f&quot;Very high count reads: {len(high_count_reads)}&quot;)

# Custom queries on collections
reads_collection = query.get_collection(&quot;Reads&quot;)
for read in reads_collection.find_iter({'read_count_category': 'very_high'}):
    print(f&quot;{read['reads_name']}: {read['reads_read_count']:,}&quot;)
</code></pre>
<h2 id="output-file-naming-conventions">Output File Naming Conventions</h2>
<p><strong>Recommended naming:</strong>
- Query results: <code>{query_type}_{threshold}_{date}.json</code>
  - Example: <code>unused_reads_50000_2025-01-14.json</code>
- Reports: <code>{analysis_type}_report_{date}.txt</code>
  - Example: <code>resource_utilization_report_2025-01-14.txt</code>
- Lineage exports: <code>{entity_type}_{entity_id}_lineage.json</code>
  - Example: <code>Assembly_Assembly0000001_lineage.json</code></p>
<h2 id="database-information">Database Information</h2>
<p><strong>Database file:</strong> <code>enigma_data.db</code> (10-13 MB)</p>
<p><strong>Collections:</strong>
- Reads: 19,307 records
- Assembly: 3,427 records
- Process: 130,560 records (provenance tracking)
- Sample: 4,119 records
- Genome: 6,688 records
- Strain: 3,106 records
- Location: 594 records
- Community: 2,140 records
- Protocol: 42 records
- OTU: 111,830 records</p>
<p><strong>Total:</strong> 281,813 records</p>
<h2 id="performance-notes">Performance Notes</h2>
<ul>
<li>Simple queries (by ID): &lt; 100ms</li>
<li>Unused reads analysis: 2-5 seconds (full scan of 19K reads + 130K processes)</li>
<li>Lineage trace: &lt; 500ms</li>
<li>Database statistics: 1-2 seconds</li>
</ul>
<h2 id="troubleshooting">Troubleshooting</h2>
<p><strong>Database not found:</strong></p>
<pre><code class="language-bash"># Create the database first
just load-store
</code></pre>
<p><strong>Query too slow:</strong></p>
<pre><code class="language-bash"># Reload database with indexes (if not already done)
rm enigma_data.db
just load-store
</code></pre>
<p><strong>Out of memory:</strong></p>
<pre><code class="language-bash"># Use lower thresholds or limit result sets
uv run python enigma_query.py unused-reads --min-count 100000  # Higher threshold = fewer results
</code></pre>
<h2 id="provenance-tracking">Provenance Tracking</h2>
<p>Every query execution is automatically tracked with complete metadata for reproducibility and auditing.</p>
<h3 id="what-is-tracked">What is Tracked</h3>
<ul>
<li><strong>Execution</strong>: Unique ID, timestamp, duration, parameters, status</li>
<li><strong>User</strong>: Username, hostname, platform</li>
<li><strong>Database</strong>: Checksum, size, modification time, record counts</li>
<li><strong>Environment</strong>: Python version, package versions</li>
<li><strong>Results</strong>: Summary statistics, output files</li>
</ul>
<h3 id="view-execution-history">View Execution History</h3>
<pre><code class="language-bash"># List all executions
uv run python query_provenance_tracker.py --list

# Generate report for specific execution
uv run python query_provenance_tracker.py --report &lt;execution_id&gt;
</code></pre>
<h3 id="provenance-records">Provenance Records</h3>
<p>All executions saved in <code>query_provenance/</code>:
- <code>YYYYMMDD_HHMMSS_querytype_execid.json</code> - Full provenance record
- <code>latest_querytype.json</code> - Most recent execution of each type</p>
<p>Each query displays its execution ID:</p>
<pre><code>ðŸ“‹ Provenance record saved: query_provenance/20251014_125537_unused_reads_1e16a3d7b455ebce.json
   Execution ID: 1e16a3d7b455ebce
</code></pre>
<h3 id="reproducing-results">Reproducing Results</h3>
<pre><code class="language-bash"># 1. Find the execution ID from output or history
uv run python query_provenance_tracker.py --list

# 2. Get the provenance record
uv run python query_provenance_tracker.py --report 1e16a3d7b455ebce

# 3. Verify database hasn't changed (check checksum)
# 4. Re-run with same parameters
</code></pre>
<h2 id="see-also">See Also</h2>
<ul>
<li><a href="../REFINED_QUERY_ANALYSIS/">REFINED_QUERY_ANALYSIS.md</a> - Refined query analysis for isolate genome reads</li>
<li><a href="../DEPLOYMENT_PROVENANCE/">DEPLOYMENT_PROVENANCE.md</a> - Complete deployment &amp; provenance guide</li>
<li><a href="../LINKML_STORE_USAGE/">LINKML_STORE_USAGE.md</a> - Detailed usage guide</li>
<li><a href="CLAUDE.md">CLAUDE.md</a> - Main project documentation</li>
</ul>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": ["content.tabs.link"], "search": "../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.92b07e13.min.js"></script>
      
    
  </body>
</html>